src/client/game.cpp

```diff
@@ -1158,31 +1163,44 @@ void Game::run()
 			&& client->checkPrivilege("fast");
 #endif
 
-	irr::core::dimension2d<u32> previous_screen_size(g_settings->getU16("screen_w"),
+	v2u32 previous_screen_size(g_settings->getU16("screen_w"),
 		g_settings->getU16("screen_h"));
 
 	while (m_rendering_engine->run()
 			&& !(*kill || g_gamecallback->shutdown_requested
 			|| (server && server->isShutdownRequested()))) {
 
-		const irr::core::dimension2d<u32> &current_screen_size =
-			m_rendering_engine->get_video_driver()->getScreenSize();
+		// Calculate dtime =
+		//    m_rendering_engine->run() from this iteration
+		//  + Sleep time until the wanted FPS are reached
+		draw_times.limit(device, &dtime);
+
+		const auto current_dynamic_info = getCurrentDynamicInfo();
+		if (!current_dynamic_info.equal(client_display_info)) {
+			client_display_info = current_dynamic_info;
+			dynamic_info_send_timer = 0.2f;
```

Comment:
0.2s is huge and i don't think the server has the necessity to be at 0.2sec, better to be 0.5 or 1sec
also i'm not fan of this coomplexity to check on each step. Instead check every <timeframe> and just send

Reply:
This is debouncing, it causes the data to only send when it stops changing. So in practice it will be 1s if the user spends 1s resizing their window. This introduces a problem, if the data is constantly changing it will never send, but I don't think this is likely on any platform